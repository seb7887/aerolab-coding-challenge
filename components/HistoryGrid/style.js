import styled from 'styled-components';

export const Grid = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-column-gap: 1.5rem;
  grid-row-gap: 2.5rem;
  padding: 1.5rem;

  @media (${props => props.theme.md}) {
    grid-template-columns: repeat(2, 1fr);
  }

  @media (${props => props.theme.sm}) {
    display: flex;
    flex-direction: column;
  }
`;
