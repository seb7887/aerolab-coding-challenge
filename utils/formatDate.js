export default function formatDate(value) {
  const getNumber = number => (number < 10 ? `0${number}` : String(number));
  const date = new Date(value);

  return `${getNumber(date.getMonth() + 1)}/${getNumber(
    date.getDate()
  )}/${getNumber(date.getFullYear())} ${getNumber(
    date.getUTCHours()
  )}:${getNumber(date.getUTCMinutes())}`;
}
