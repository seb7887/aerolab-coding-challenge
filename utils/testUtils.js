import faker from 'faker';

export const fakeUser = {
  name: 'John Kite',
  points: '7000'
};

export const mockCategories = [
  'all',
  'gaming',
  'laptops',
  'tablets',
  'cameras',
  'audio',
  'tv',
  'drones',
  'accessories',
  'home'
];

export const fakeProduct = {
  _id: 7,
  name: 'iPhone',
  category: 'Phones',
  cost: 800,
  img: {
    url: '/static/image/image.png'
  }
};

export const generateProduct = () => {
  const newProduct = {
    _id: faker.random.uuid(),
    name: faker.commerce.productName(),
    category: faker.commerce.productAdjective(),
    cost: 800,
    img: {
      url: faker.image.imageUrl()
    }
  };
  return newProduct;
};

export const fakeProducts = [fakeProduct];

export const generateProducts = quantity => {
  let products = [];

  for (let i = 0; i < quantity; i++) {
    const newProduct = generateProduct();
    products.push(newProduct);
  }

  return products;
};
