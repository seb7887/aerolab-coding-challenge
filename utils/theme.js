const black = '#000';
const white = '#fff';
const cyan = '#0ad4fa';
const orange = '#ff6e05';
const red = 'red';
const green = 'green';

const bgLight = '#e9ebf1';
const bgDark = '#212531';
const secondaryBgLigth = 'rgba(0, 0, 0, 0.1)';
const secondaryBgDark = 'rgba(255, 255, 255, 0.1)';
const containerBgLight = '#fff';
const containerBgDark = '#000';
const boxShadowLight = '0px 0px 3px 0px rgba(0, 0, 0, 0.1)';
const boxShadowDark = 'rgba(31, 31, 31, 0.1)';
const productShadowLight = '1px 4px 5px 1px rgba(0, 0, 0, 0.1)';
const productShadowDark = '1px 4px 5px 1px rgba(255, 255, 255, 0.1)';

const primaryLight = '#616161';
const primaryDark = '#f8f8f8';

const secondaryLight = black;
const secondaryDark = white;
const greyLight = '#e0e0e0';
const greyDark = '#a3a3a3';

const theme = {
  light: {
    black,
    white,
    cyan,
    orange,
    red,
    green,
    grey: greyLight,
    bg: bgLight,
    secondaryBg: secondaryBgLigth,
    containerBg: containerBgLight,
    boxShadow: boxShadowLight,
    productShadow: productShadowLight,
    primary: primaryLight,
    secondary: secondaryLight
  },
  dark: {
    black,
    white,
    cyan,
    orange,
    red,
    green,
    grey: greyDark,
    bg: bgDark,
    secondaryBg: secondaryBgDark,
    containerBg: containerBgDark,
    boxShadow: boxShadowDark,
    productShadow: productShadowDark,
    primary: primaryDark,
    secondary: secondaryDark
  }
};

export default theme;
