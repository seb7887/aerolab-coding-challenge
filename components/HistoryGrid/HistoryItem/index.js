import React from 'react';
import PropTypes from 'prop-types';

import formatDate from '../../../utils/formatDate';

import { Container, Img } from './style';

const HistoryItem = ({ item }) => (
  <Container data-testid='history-item'>
    <div>
      <span data-testid='create-date'>{formatDate(item.createDate)}</span>
    </div>
    <Img src={item.img.url} alt={item.name} />
    <div className='description'>
      <span data-testid='category'>{item.category}</span>
      <span className='name' data-testid='name'>
        {item.name}
      </span>
    </div>
  </Container>
);

HistoryItem.propTypes = {
  item: PropTypes.object.isRequired
};

export default HistoryItem;
