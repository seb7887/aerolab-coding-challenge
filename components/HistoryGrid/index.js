import React from 'react';

import Banner from '../Banner';
import HistoryItem from './HistoryItem';

import { Grid } from './style';

const History = ({ history }) => (
  <>
    <Banner>History</Banner>
    <Grid>
      {history.map(item => {
        return <HistoryItem key={item._id} item={item} />;
      })}
    </Grid>
  </>
);

export default History;
