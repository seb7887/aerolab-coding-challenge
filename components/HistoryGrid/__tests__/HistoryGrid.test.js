import { render } from 'react-testing-library';

import HistoryGrid from '../index';

import { fakeHistoryItems } from '../../../utils/testUtils';

describe('HistoryGrid', () => {
  it('renders and matches snapshot', () => {
    const { asFragment } = render(<HistoryGrid history={fakeHistoryItems} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display a correct quantity of products', () => {
    const { getAllByTestId } = render(
      <HistoryGrid history={fakeHistoryItems} />
    );

    expect(getAllByTestId('history-item').length).toBe(fakeHistoryItems.length);
  });
});
