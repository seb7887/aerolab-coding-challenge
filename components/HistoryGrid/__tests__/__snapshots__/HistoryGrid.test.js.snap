// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryGrid renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="style__Content-ya8ux5-0 fnKozQ"
  >
    <h1
      class="title"
    >
      History
    </h1>
    <div
      class="overlay"
    />
  </div>
  <div
    class="style__Grid-sc-1yaakt4-0 cDtyHz"
  >
    <div
      class="style__Container-sc-1ba92t9-0 dVxYRm"
      data-testid="history-item"
    >
      <div>
        <span
          data-testid="create-date"
        >
          04/24/2019 22:17
        </span>
      </div>
      <img
        alt="iPhone"
        class="style__Img-sc-1ba92t9-1 bIOfTM"
        src="/static/image/image.png"
      />
      <div
        class="description"
      >
        <span
          data-testid="category"
        >
          Phones
        </span>
        <span
          class="name"
          data-testid="name"
        >
          iPhone
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
